lib/keystore/helpers.go

```diff
@@ -112,11 +112,19 @@ func GenerateKeypair(keytype string, kp crypto.Keypair, basepath string, passwor
 }
 
 // LoadKeystore loads a new keystore and inserts the test key into the keystore
-func LoadKeystore(key string, ks Keystore) error {
+func LoadKeystore(key string, ks TyperInserter) (err error) {
 	if key != "" {
-
-		var kr Keyring
-		var err error
+		var kr interface {
+			Alice() crypto.Keypair
+			Bob() crypto.Keypair
+			Charlie() crypto.Keypair
+			Dave() crypto.Keypair
+			Eve() crypto.Keypair
+			Ferdie() crypto.Keypair
+			George() crypto.Keypair
+			Heather() crypto.Keypair
+			Ian() crypto.Keypair
+		}
```

Comment:
would suggest to refactor this function to:

```go
type dummyKeyRing interface {
  Alice() crypto.Keypair
  Bob() crypto.Keypair
  Charlie() crypto.Keypair
  Dave() crypto.Keypair
  Eve() crypto.Keypair
  Ferdie() crypto.Keypair
  George() crypto.Keypair
  Heather() crypto.Keypair
  Ian() crypto.Keypair
}
type keyRingProvider func() (kr dummyKeyRing)
func LoadKeystore(key string, ks TyperInserter, provider keyRingProvider) (err error) {
  if key == "" { return nil }
  kr := provider()
  switch strings.ToLower(key) {
  ...
  
}
// in the caller side

LoadKeystore(..., ..., NewEd25519Keyring) // or LoadKeystore(..., ..., NewSr25519Keyring)
```

Reply:
Done in #2886 since this is a bit out of scope and deltas are not that small :wink: